homeassistant/util/dt.py

```diff
@@ -340,7 +340,7 @@ def find_next_time_expression_time(
             #  2. 2021.10.31 02:00:00+01:00 with pattern 02:30 -> 2021.10.31 02:30:00+01:00
             return result.replace(fold=now.fold)
 
-        if now_is_ambiguous and now.fold == 0 and not result_is_ambiguous:
```

Comment:
Guard above ensures `now_is_ambiguous` is always true

Guard above ensures `result_is_ambiguous` is else false